src/components/Carousel.tsx

```diff
@@ -1,26 +1,103 @@
 import React from 'react';
 import './Carousel.scss';
+import classNames from 'classnames';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li>
-        <img src="./img/1.png" alt="1" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="2" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="3" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="4" />
-      </li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite?: boolean;
+};
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [slide, setSlide] = React.useState(0);
+
+  const totalItems = images.length;
+  const slideItems = Math.max(Math.ceil((totalItems - frameSize) / step), 0);
+
+  const nextSlide = infinite || slide < slideItems;
+  const previousSlide = infinite || slide > 0;
+
+  const containerStyle = React.useMemo(
+    () => ({
+      width: `${itemWidth * frameSize}px`,
+    }),
+    [itemWidth, frameSize],
+  );
+
+  const listStyle = React.useMemo(() => {
+    const maxTranslateX = (totalItems - frameSize) * itemWidth;
+    const translateX = Math.min(slide * itemWidth * step, maxTranslateX);
+
+    return {
+      transform: `translateX(-${translateX}px)`,
+      transition: `transform ${animationDuration}ms ease-in-out`,
+    };
+  }, [totalItems, frameSize, itemWidth, slide, step, animationDuration]);
+
+  const handleNextClick = () => {
+    if (nextSlide) {
+      setSlide(currentSlide => (currentSlide + 1) % (slideItems + 1));
+    }
+  };
+
+  const handlePrevClick = () => {
+    if (previousSlide) {
+      setSlide(currentSlide =>
+        currentSlide === 0 ? slideItems : currentSlide - 1,
+      );
+    }
+  };
+
+  return (
+    <div className="Carousel">
+      <div className="Carousel__container" style={containerStyle}>
+        <ul className="Carousel__list" style={listStyle}>
+          {images.map((image, index) => (
+            <li className="Carousel__item" key={index}>
+              <img src={image} alt={`Image ${index + 1}`} width={itemWidth} />
+            </li>
+          ))}
+        </ul>
+      </div>
+
+      <div className="Carousel__navigate">
+        <button
+          className={classNames([
+            'Carousel__button',
+            !previousSlide && !infinite ? 'Carousel__button--disable' : '',
+          ])}
```

Comment:
```suggestion
          className={classNames('Carousel__button', {
            'Carousel__button--disable': !previousSlide && !infinite,
          })}
```
correct classNames usage in this case